linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -511,15 +512,30 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
         case other                                        => other
       }
 
-      js.Function(arrow && useArrowFunctions, newParams,
-          js.Block(extractRestParam, cleanedNewBody))
+      val actualArrowFun = arrow && useArrowFunctions
+      val normalParams = params.map(transformParamDef(_, rest = false))
+
+      restParam match {
+        case None =>
+          js.Function(actualArrowFun, normalParams, cleanedNewBody)
+
+        case Some(restParam) if esFeatures.useECMAScript2015 =>
+          val params =
+            normalParams :+ transformParamDef(restParam, rest = true)
+
+          js.Function(actualArrowFun, params, cleanedNewBody)
```

Comment:
Have you considered also applying the same refactoring to the `javascript.Tree`s? It seems like it would be more consistent and cleaner.

Reply:
Yep. I'll do that (likely in a separate commit).